### PR TITLE
removed need for reset pin

### PR DESCRIPTION
**Pull request by tkoester**
The MFRC522 library insists on configuring a reset pin. According to the data sheet, this should only be neccessary when one wants to "hard" power down the transceiver. Soft power down seems not to be affected.

Since  I'm running out of pins on my design, I figured I wouldn't need the reset connection to the MFRC522. I don't know yet if this proves to be stable in day-to-day use, though. Your mileage may vary.

In a nutshell, this code uses a soft reset on PCD_Init() not only if the device has already been out of hard reset / power-down, but also if the reset pin has not been configured.

### CODE REVIEW

File: MFRC522.cpp [hunk 1]
```diff
@@ -13,7 +13,7 @@
 /**
  * Constructor.
  */
-MFRC522::MFRC522() {
```
Comment (Rotzbua) [inline MFRC522.cpp]: _resetPowerDownPin is type `byte` which is `uint8_t`, so `-1` is not so good idea. 

Code the old code is correct .

File: MFRC522.cpp [hunk 2]
```diff
@@ -197,19 +197,25 @@ MFRC522::StatusCode MFRC522::PCD_CalculateCRC(	byte *data,		///< In: Pointer to
...
 void MFRC522::PCD_Init() {
+	bool hardReset = false;
+
 	// Set the chipSelectPin as digital output, do not select the slave yet
 	pinMode(_chipSelectPin, OUTPUT);
 	digitalWrite(_chipSelectPin, HIGH);
 	
-	// Set the resetPowerDownPin as digital output, do not reset or power down.
-	pinMode(_resetPowerDownPin, OUTPUT);
+	if (_resetPowerDownPin != -1) {
```
Comment (Rotzbua) [inline MFRC522.cpp]: Better use `!= 0xFF` and define the default of _reset..  as 0xFF

**Comment by Rotzbua** [comment]
Pls add comment where you reference to the pages in doc.

**Comment by tkoester** [comment]
Regarding documentation:

According to NXP's MFRC522 data sheet Rev. 3.9 (27 April 2016), section 8.6.1, NRSTPD enables hard power-down mode. According to section 8.6.2, soft power-down mode can be entered via CommandReg. Obviously, there is also a SoftReset command (see section 10.3.1.10), as well as a power-on reset (see e.g. Fig. 2).
It seems to me that every conceivable use case of NRSTPD should therefore be covered by an appropriate register setting / command.

**Comment by Rotzbua** [comment]
@tkoester Sry, I mean documentation in the source code. My experience is that otherwise somebody will change it later because he did not know about the specification.

**Comment by tkoester** [comment]
@Rotzbua Sorry I misunderstood you there.
I added some documentation to the source code. And I sure hope I got Git right by now - I've only ever really used centralized VCS until now.
